lib/core/connector_settings.rb

```diff
@@ -98,6 +96,20 @@ def run_ml_inference?
       return_if_present(@elasticsearch_response.dig(:pipeline, :run_ml_inference), @connectors_meta.dig(:pipeline, :default_run_ml_inference), DEFAULT_RUN_ML_INFERENCE)
     end
 
+    def formatted
+      properties = ["ID: #{id}"]
+      properties << "Service type: #{service_type}" if service_type
+      "connector (#{properties.join(', ')})"
+    end
+
+    def needs_service_type?
+      service_type.to_s.strip.empty?
```

Comment:
Hmmm, why `to_s.strip`? is it possible to be something like " " or not be a string?

Reply:
Yes, to prevent something like `" "`, maybe it's over-defensive?

Reply:
I'm on the fence here - I like code that can recover from failure, on the other hand I like it when the code raises problems as fast as possible in the development phase.

I think it's okay to keep it this way here, but also to think how many errors or invalid data we can recover from in the whole repo.